pandas/core/indexes/base.py

```diff
@@ -4577,21 +4577,23 @@ def get_value(self, series, key):
         # use this, e.g. DatetimeIndex
         # Things like `Series._get_value` (via .at) pass the EA directly here.
         s = getattr(series, "_values", series)
-        if isinstance(s, (ExtensionArray, Index)) and is_scalar(key):
-            # GH 20882, 21257
-            # Unify Index and ExtensionArray treatment
-            # First try to convert the key to a location
-            # If that fails, raise a KeyError if an integer
-            # index, otherwise, see if key is an integer, and
-            # try that
-            try:
-                iloc = self.get_loc(key)
-                return s[iloc]
-            except KeyError:
-                if len(self) > 0 and (self.holds_integer() or self.is_boolean()):
-                    raise
-                elif is_integer(key):
-                    return s[key]
+        if isinstance(s, ExtensionArray):
+            if is_scalar(key):
+                # GH 20882, 21257
+                # First try to convert the key to a location
+                # If that fails, raise a KeyError if an integer
+                # index, otherwise, see if key is an integer, and
+                # try that
+                try:
+                    iloc = self.get_loc(key)
+                    return s[iloc]
+                except KeyError:
+                    if len(self) > 0 and (self.holds_integer() or self.is_boolean()):
+                        raise
+                    elif is_integer(key):
+                        return s[key]
+            else:
+                raise InvalidIndexError(key)
```

Comment:
The actual change is that I raise an error here when the key is not a scalar (which would happen anyway below otherwise, I think, as `_engine.get_value` is meant for scalar indices).

Reply:
>  I think, as _engine.get_value is meant for scalar indices).

Might be worth renaming something like get_value_scalar (not for this PR)

Reply:
you don't need the else here, can you add the issue number and short comment

Reply:
Added a comment to explain